### Tugas 2 - 5 Cross Validation | KOMB
### Nama: Sofirul Danatriya
### NIM	: 20/455453/PA/19668

In [1]:
# import library
import pandas as pd

# load file
df = pd.read_excel("Tugas3-ML.xlsx")

# menghilangkan id
df = df.iloc[5:,1:]
df.columns = ['Real Label', 'Prediction']

df.head()

Real Label Prediction
5          1          1
6          1          1
7          1          0
8          1          0
9          1          0

In [2]:
# fungsi untuk membuat confussion matrix
def generateConfussionMatrix(dataframe):
    (TP, FP, TN, FN) = (0, 0, 0, 0)
    for i, row in dataframe.iterrows():
        if row['Real Label'] == 1 and row['Prediction'] == 1:
            TP += 1
        elif row['Real Label'] == 1 and row['Prediction'] == 0:
            FN += 1
        elif row['Real Label'] == 0 and row['Prediction'] == 1:
            FP += 1
        elif row['Real Label'] == 0 and row['Prediction'] == 0:
            TN += 1

    return (TP, FP, TN, FN)

# Fungsi untuk menghitung Akurasi Presisi dan Recall
def hitungAPR(TP, FP, TN, FN):
    akurasi = (TP + TN) / (TP + TN + FP + FN)
    presisi = (TP) / (TP + FP)
    recall = (TP) / (TP + FN)
    return (akurasi, presisi, recall)

In [3]:
# menghitung confussion matrix
(TP, FP, TN, FN) = generateConfussionMatrix(df)
print('Predicted\t       ActualValues')
print('\t\t\tPositive\tNegative')
print(f'Positive\t{TP}\t\t\t{FP}')
print(f'Negative\t{FN}\t\t\t{TN}')

Predicted	       ActualValues
			Positive	Negative
Positive	16			21
Negative	10			19


In [4]:
# Menghitung Akurasi, Presisi, dan recall/sensitivitas
(akurasi, presisi, sensitivitas) = hitungAPR(TP, FP, TN, FN)

print(f'Akurasi\t\t: {round(akurasi*100,2)}%')
print(f'Presisi\t\t: {round(presisi*100,2)}%' )
print(f'Sensitivitas\t: {round(sensitivitas*100,2)}%', )



Akurasi		: 53.03%
Presisi		: 43.24%
Sensitivitas	: 61.54%


Dari Perhitungan diatas, kita dapat diambil kesimpulan bahwa:
Mesin memiliki akurasi 53.03%, presisi 43.24%, dan recall/sensitivitas 61.54%.